kx/kx.go

```diff
@@ -0,0 +1,104 @@
+package kx
+
+import (
+	"crypto/rand"
+	"errors"
+	"golang.org/x/crypto/blake2b"
+	"golang.org/x/crypto/curve25519"
+)
+
+const SeedBytes = 32
+const SecretKeyBytes = 32
+const PublicKeyBytes = 32
+const SessionKeyBytes = 32
+const scalarMultBytes = 32
+
+var invalidSeed = errors.New("invalid seed")
+var cryptoError = errors.New("crypto error")
+var notImplemented = errors.New("not implemented")
+
+type KeyPair struct {
+	pk []byte
+	sk []byte
+}
+
+func NewKeyPair(seed []byte) (*KeyPair, error) {
```

Comment:
needs some doc. if nothing else jsut that passing nil is ok and what happens when it's nil

Reply:
changed to make seeded key generation internal operation, used only for testing